design/gateway-apis-implementation.md

```diff
@@ -264,16 +294,7 @@ Precedence rules in the spec must also be followed.
 The rules for implementing RouteBindingSelector are straightforward and will be implemented per the spec.
 #### Gateway Status
 
-The only status that Contour will update in the Gateway is the `status.[]ListenerStatus` slice.
-This slice contains `ListenerStatus` entries that are expected to be keyed by `port`.
-If more than one Listener shares the same port, the `ListenerStatus` reports the combined status.
-
-As the API currently stands, Contour must combine the statuses for all Listeners that share a `port`, including Listeners that are being merged.
-If this proves awkward, we may need to ask for some upstream changes.
-
-Whenever Contour looks at a Listener, it will add an `Admitted` Condition.
-In the event that a ListenerStatus has everything okay (all mergeable Listeners pass validation), the `Admitted` condition will be `status: true`.
-If there are any errors, the `Admitted` Condition will be `status: false`, and the Reason will tell you more about why.
+The gateway controller is responsible for managing Gateway status.
```

Comment:
@skriss I'm trying to generalize the design to use "gateway controller" since it appears that the maintainer team has not achieved consensus on where the Gateway controller should live. A potential approach to unblock us, is to support multiple [controller](https://github.com/kubernetes-sigs/gateway-api/blob/v0.2.0/apis/v1alpha1/gatewayclass_types.go#L49-L63) domain/paths. 

__Option 1- Operator Managed:__ The operator will own a Gateway of class "projectcontour.io/contour-operator" and be responsible for managing the Gateway, including status. Contour will be responsible reading Gateway status to manage HTTPRoutes and TLSRoutes, including status. 

__Option 2- Contour Managed:__ The operator will __not__ manage a Gateway of class "projectcontour.io/contour". Instead, Contour will be responsible for managing the Gateway, including status. The GatewayClass should either not reference an object or reference an object other than the Contour custom resource... maybe a ConfigMap. With this approach, users are expected to manage Contour using standard Kubernetes manifests. Contour will need to ensure the following for a Gateway:
- If [addresses](https://github.com/kubernetes-sigs/gateway-api/blob/v0.2.0/apis/v1alpha1/gateway_types.go#L115-L132) are specified, ensure they match Envoy service addresses and surface the appropriate Gateway status condition if not.
- Ensure each [listener port/protocol](https://github.com/kubernetes-sigs/gateway-api/blob/v0.2.0/apis/v1alpha1/gateway_types.go#L163-L184) matches the port name/protocol in the Envoy service and surface the appropriate Gateway/Listener status condition if not.

Supporting both approaches will be a short-term approach that allows us to figure out the model that works best for Contour. Thoughts? cc: @stevesloka @sunjayBhatia @youngnick

Reply:
While I agree that having something that doesn't require installing two things would be preferable, we need to not be pausing development on Gateway support while we figure out what that looks like. What we have in this current design is deliverable with what we currently have, and I would rather have something working first before spending a lot of engineering effort refactoring.